### PR TITLE
Adding structures to support New Security Type

### PR DESCRIPTION
**Pull request by shayan-msft**
adding types to support new security type

### CODE REVIEW

File: rpc/nodeagent/compute/moc_nodeagent_virtualmachine.pb.go [hunk 1]
```diff
@@ -1287,115 +1295,117 @@ func init() {
...
-	0xf7, 0xe3, 0x99, 0xd4, 0x57, 0xf3, 0xc7, 0xf7, 0xb4, 0x4e, 0x92, 0xe3, 0x7e, 0x0f, 0xbd, 0x86,
-	0x85, 0xa4, 0xc9, 0x28, 0xfa, 0xe2, 0x5e, 0xd8, 0xe2, 0x28, 0x69, 0x7e, 0xf9, 0xb6, 0xb0, 0xf1,
-	0xda, 0xc7, 0xb0, 0xb6, 0xdf, 0xa7, 0xde, 0xcd, 0x59, 0xe6, 0x9f, 0x75, 0x68, 0x73, 0xaa, 0xff,
-	0x0e, 0x83, 0x48, 0x8d, 0x9a, 0x1f, 0x98, 0x65, 0xb2, 0xa6, 0x19, 0x4f, 0x7f, 0x04, 0x98, 0xa4,
-	0x1b, 0xdd, 0x8f, 0xd1, 0x54, 0x73, 0x36, 0x7f, 0xf2, 0xd6, 0xb8, 0x94, 0xc0, 0xde, 0x67, 0xdf,
-	0xb4, 0x7a, 0x4c, 0xf5, 0xe3, 0x2b, 0x0d, 0xda, 0x09, 0x98, 0x27, 0xb8, 0xe4, 0xd7, 0x6a, 0x27,
-	0xe0, 0xde, 0x8e, 0x88, 0xbc, 0x9d, 0xb1, 0xd3, 0x1d, 0xeb, 0xf4, 0x6a, 0xde, 0xc4, 0xf7, 0xfc,
-	0x7f, 0x01, 0x00, 0x00, 0xff, 0xff, 0x22, 0xb5, 0xde, 0x7e, 0x67, 0x15, 0x00, 0x00,
+	// 1752 bytes of a gzipped FileDescriptorProto
+	0x1f, 0x8b, 0x08, 0x00, 0x00, 0x00, 0x00, 0x00, 0x02, 0xff, 0xb4, 0x58, 0xdd, 0x6e, 0x23, 0xb7,
```
Comment (SethBe) [inline rpc/nodeagent/compute/moc_nodeagent_virtualmachine.pb.go:1299]: >x00, 0x00, 0x00, 0x00, 0x0 [](http://example.com/codeflow?start=20&length=26)

not sure what this is?

File: rpc/common/moc_common_computecommon.proto [hunk 2]
```diff
@@ -172,10 +172,16 @@ message VirtualMachineRunCommandInstanceView {
 	string Error = 4;
 }
 
+enum ConfidentialVMHardwareType {
+	HardwareTypeNone = 0;
+	HardwareTypeSEVSNP = 1;
+}
+
 // describes the the capabilities of this node in terms of what type of VM can be created
 // In the future when CVM comes, another bool will be added to check hardware capability.
```
Comment (SethBe) [inline rpc/common/moc_common_computecommon.proto:181]: >hen CVM comes [](http://example.com/codeflow?start=18&length=13)

remove

File: rpc/nodeagent/compute/moc_nodeagent_virtualmachine.pb.go (hunk 1 above)
Comment (shayan-msft) [reply rpc/nodeagent/compute/moc_nodeagent_virtualmachine.pb.go:1299]: I think this is the binary compressed protobuf. It gets autogenerated each time I make, pretty sure we need to check this in

**Review by danmamsft — APPROVED** [review]

**Review by danmamsft — APPROVED** [review]

**Review by msftpgayam — APPROVED** [review]

**Review by danmamsft — APPROVED** [review]

**Review by msftpgayam — APPROVED** [review]

**Comment by SethBe** [comment]
[heart]         Seth Beinhart reacted to your message:
________________________________

**Pull request merged**